ui/inspectors/service.reel/dyndns-service.reel/dyndns-service.js

```diff
@@ -0,0 +1,26 @@
+var Component = require("montage/ui/component").Component,
+    ServiceDyndnsProvider = require("core/model/enumerations/service-dyndns-provider").ServiceDyndnsProvider;
+
+/**
+ * @class DynamicDnsService
+ * @extends Component
+ */
+exports.DynamicDnsService = Component.specialize({
+    providerOptions: {
+        value: null
+    },
+    
+    templateDidLoad: {
+        value: function() {
+          this.providerOptions = ServiceDyndnsProvider.members;
+        }
+    },
+
+    save: {
+        value: function() {
+            if (this.object.password === "" || this.object.password === null) {
+                delete this.object.password;
```

Comment:
You should just set the property to null if it's an empty string, montage-data mapping will take care of not sending it to the middleware.
As a rule of thumb, we shouldn't remove properties from a dataObject that are part of its schema.